crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -347,3 +352,184 @@ async fn cow_amm(web3: Web3) {
+        usdc.approve(onchain.contracts().allowance, to_wei_with_exp(1000, 6))
+    );
+
+    // spawn a mock solver so we can later assert things about the received auction
+    let mock_solver = Mock::default();
+    colocation::start_driver(
+        onchain.contracts(),
+        vec![
+            SolverEngine {
+                name: "test_solver".into(),
+                account: solver.clone(),
+                endpoint: colocation::start_baseline_solver(onchain.contracts().weth.address())
+                    .await,
+            },
+            SolverEngine {
+                name: "mock_solver".into(),
+                account: solver.clone(),
+                endpoint: mock_solver.url.clone(),
+            },
+        ],
+        colocation::LiquidityProvider::UniswapV2,
+    );
+    let services = Services::new(onchain.contracts()).await;
+    services
+        .start_autopilot(
+            None,
+            vec![
+                "--drivers=test_solver|http://localhost:11088/test_solver,mock_solver|http://localhost:11088/mock_solver".to_string(),
+                "--price-estimation-drivers=test_solver|http://localhost:11088/test_solver"
+                    .to_string(),
+            ],
+        )
+        .await;
+    services
+        .start_api(vec![
+            "--price-estimation-drivers=test_solver|http://localhost:11088/test_solver".to_string(),
+        ])
+        .await;
+
+    // Place Orders
+    let order = OrderCreation {
+        sell_token: usdc.address(),
+        sell_amount: to_wei_with_exp(1000, 6),
+        buy_token: usdt.address(),
+        buy_amount: to_wei_with_exp(2000, 6),
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(trader.private_key()).unwrap()),
+    );
+
+    // Warm up co-located driver by quoting the order (otherwise placing an order
+    // may time out)
+    let _ = services
+        .submit_quote(&OrderQuoteRequest {
+            sell_token: usdc.address(),
+            buy_token: usdt.address(),
+            side: OrderQuoteSide::Sell {
+                sell_amount: SellAmount::BeforeFee {
+                    value: to_wei_with_exp(1000, 6).try_into().unwrap(),
+                },
+            },
+            ..Default::default()
+        })
+        .await;
+
+    let order_id = services.create_order(&order).await.unwrap();
+    let limit_order = services.get_order(&order_id).await.unwrap();
+    assert_eq!(limit_order.metadata.class, OrderClass::Limit);
+
+    // Drive solution
+    tracing::info!("Waiting for trade.");
+    wait_for_condition(TIMEOUT, || async {
+        // Keep mining blocks to trigger the event indexing logic
+        onchain.mint_block().await;
+        tokio::time::sleep(tokio::time::Duration::from_millis(1_000)).await;
```

Comment:
why do we wait 1 seconds, shouldn't it wait 12 seconds for the next block? :thinking:

Reply:
In a forked test there is not auto mining of blocks. Blocks get mined whenever we execute a transaction on the fork so the mainnet block time is not really relevant here.